flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -143,36 +144,39 @@ class FIRToLLVMTypeConverter : public mlir::LLVMTypeConverter {
     return mlir::LLVM::LLVMType::getInt64Ty(llvmDialect);
   }
 
+  // TODO
+  bool requiresExtendedDesc() { return false; }
+
   // This corresponds to the descriptor as defined ISO_Fortran_binding.h and the
   // addendum defined in descriptor.h.
-  // FIXME: This code should be generated and follow SPOT
-  mlir::LLVM::LLVMType convertBoxType(fir::BoxType box) {
+  mlir::LLVM::LLVMType convertBoxType(fir::BoxType box, int rank = -1) {
     // (buffer*, ele-size, rank, type-descriptor, attribute, [dims])
     SmallVector<mlir::LLVM::LLVMType, 6> parts;
     mlir::Type ele = box.getEleTy();
-    // auto *ctx = box.getContext();
     auto eleTy = unwrap(convertType(ele));
     // buffer*
     if (ele.isa<fir::SequenceType>() && eleTy.isPointerTy())
       parts.push_back(eleTy);
```

Comment:
Won't it be a problem for some interfaces that accepts descriptor of all types to have the buffer type explicit ?

Reply:
I flipped a coin and left this. In practice, since this is a memory reference, the type isn't too important. Experience may prove out that using an `i8*` would be better, though. Not sure yet.